cpu/stm32f4/periph/spi.c

```diff
@@ -102,6 +111,9 @@ int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
             return -2;
     }
 
+    /* initialize lock */
+    mutex_init(&spi_config[dev].lock);
```

Comment:
@PeterKietzmann Moving it to board.c (and a common configuration in board.h or periph_conf.h) would be the logical place to have it since when you are designing the hardware and application you will have to decide on a common subset of the available settings: we have to obey the requirements of the slowest device, we need to find a common clock polarity configuration that works on all of the connected devices etc.